apps/api/src/routers/admin.py

```diff
@@ -0,0 +1,112 @@
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Optional
+
+from fastapi import APIRouter, Body, Depends, HTTPException, status
+from pydantic import BaseModel, Field, TypeAdapter, ValidationError
+
+from auth.authorization import require_role
+from auth.user_identity import User, utc_now
+from models.ApplicationData import Decision, Review
+from services import mongodb_handler
+from services.mongodb_handler import BaseRecord, Collection
+from utils.user_record import Applicant, Role
+
+log = getLogger(__name__)
+
+router = APIRouter()
+
+ADMIN_ROLES = (Role.DIRECTOR, Role.REVIEWER)
+
+
+class ApplicationDataSummary(BaseModel):
+    first_name: str
+    last_name: str
+    school: str
+    submission_time: datetime
+
+
+class ApplicantSummary(BaseRecord):
+    uid: str = Field(alias="_id")
+    status: str
+    decision: Optional[Decision] = None
+    application_data: ApplicationDataSummary
+
+
+@router.get("/applicants")
+async def applicants(
+    user: User = Depends(require_role(ADMIN_ROLES)),
+) -> list[ApplicantSummary]:
+    """Get records of all applicants."""
+    log.info("%s requested applicants", user)
+
+    records: list[dict[str, object]] = await mongodb_handler.retrieve(
+        Collection.USERS,
+        {"role": Role.APPLICANT},
+        [
+            "_id",
+            "status",
+            "application_data.first_name",
+            "application_data.last_name",
+            "application_data.school",
+            "application_data.submission_time",
+            "application_data.reviews",
+        ],
+    )
+
+    for record in records:
+        _include_review_decision(record)
+
+    try:
+        return TypeAdapter(list[ApplicantSummary]).validate_python(records)
+    except ValidationError:
+        raise RuntimeError("Could not parse applicant data.")
+
+
+@router.get("/applicant/{uid}")
+async def applicant(
+    uid: str,
+) -> Applicant:
```

Comment:
Issue: this endpoint does not enforce admin authorization and could be accessed by anyone. We'll need to include the `require_role` dependency.

Reply:
Partly my fault from last year: we can [Add `dependencies` to the _path operation decorator_](https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-in-path-operation-decorators/#add-dependencies-to-the-path-operation-decorator) instead of as a function parameter to avoid confusion that the parameter is unused.